conbench/entities/benchmark_result.py

```diff
@@ -130,62 +134,54 @@ def create(userres):
-                # repo_url_result = userres["github"]["repository"] if
-                # repo_url_run != repo_url_result: raise
-                #     BenchmarkResultValidationError( f"Result refers to
-                #         repository URL '{repo_url_result}', but Run
-                #         '{run.id}' " f"refers to repository URL
-                #     '{repo_url_run}'" )
+        validate_and_augment_result_tags(userres)
+        validate_run_result_consistency(userres)
+
+        result_error_for_db: Optional[Dict] = None
+
+        # Handle 'error' scenarios. Obvious: user-given error object set.
+        if "error" in userres:
+            # `data["error"]` is a JSON object.
+            result_error_for_db = userres["error"]
+
+        # Now, check for a more subtle error condition, based on the per-
+        # iteration samples. Rely on `stats` key to exist (see checks above).
+        if "stats" in userres and do_iteration_samples_look_like_error(
+            userres["stats"]["data"]
+        ):
+            if "error" not in userres:
+                # User did not set `error`, but the number sequence indicates
+                # that this result is to be considered 'errored'. Set a generic
+                # error message.
+                result_error_for_db = {
+                    "status": "Partial result: not all iterations completed"
+                }
+
+        # This is going to be the structure based on which we will do
+        # DB insertion.
+        benchmark_result_data = {}
+        benchmark_result_data["error"] = result_error_for_db
 
         # Temporary: keep name `data` -- it's unfortunate that we have the name
         # `data` here while also having key(s) in the dict(s) that are called
         # `data`. Assign a recognizable name to the big, outest, user-given
         # object: userres is not ideal, but a start.
         data = userres
+
+        # At this point, assume that data["tags"] is a flat dictionary with
+        # keys being non-empty strings, and values being non-empty strings.
         tags = data["tags"]
         has_error = "error" in data
         has_stats = "stats" in data
 
-        # defaults
-        benchmark_result_data = {}
-
-        # Note(JP): if no data was provided, should this stay `True`?
-        # We will see. https://github.com/conbench/conbench/issues/803.
-        any_errored_iteration = True
-
         if has_stats:
             # Note(JP): this is the result of marshmallow
             # deserializaiton/validation against the `BenchmarkResultCreate`
             # schema below. This needs better naming. Most importantly, we
             # should make use of typing information derived from the schema.
-            benchmark_result_data = data["stats"].copy()
+            # Merge this on top of the above-defined benchmark_result_data.
+            benchmark_result_data = benchmark_result_data | data["stats"].copy()
```

Comment:
haha, I still have to get used to this notation

Reply:
>  I still have to get used to this notation

Me, too! It's only the second time I have used that. But most times before in my Python life when I did `d.update(d2)`, I felt that I don't want to modify an existing dictionary in-place, and there must be a nicer way to merge dictionaries into a new object. I think I'll get used to the pipy pipe | : )